community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/OnlineIndexProxy.java

```diff
@@ -97,7 +98,10 @@ public void start()
     @Override
     public IndexUpdater newUpdater( final IndexUpdateMode mode )
     {
-        IndexUpdater actual = accessor.newUpdater( forcedIdempotentMode ? IndexUpdateMode.RECOVERY : mode );
+        // If this proxy is flagged with taking extra care about idempotency then escalate ONLINE to ONLINE_IDEMPOTENT.
+        // At the time of writing this there's no other mode that needs to be escalated.
+        IndexUpdater actual = accessor.newUpdater(
```

Comment:
i think this condition moving us in the wrong direction and its a step that puts into the valley of assumptions based bugs even deeper then it was before. Just looking into what i can observe from this method signature and code around it its an accessor who should be able to create appropriate updater, based on correctly supplier accessor factory

Reply:
I don't think it makes things worse, it effectively does the exact same thing it used to do with the added explicitness around more index update modes. I can see if it'd be possible to makes this even better tho.